File: lbrtrn2tsk4.py
def main():
    N = int(input('Введите кол-во спортсменов: '))
    maxFml = ''
    maxRsltt = 100000
    for i in range(N):
        
        print('Введите результат и фамилию спортсмена:')
        lchshrsltt = float(input()) #res 1
        fmlpbdtl = input() # surname 1

        if (lchshrsltt < maxRsltt):
            maxRsltt = lchshrsltt 
            maxFml = fmlpbdtl

    print('Победил', maxFml, 'с результатом:', maxRsltt)

main()

File: lbrtrn2tsk4.pas
program lbrtrn2tsk4;
var N, i: integer; lchshrsltt, maxRsltt: Real; maxFml, fmlpbdtl: string;
begin
  writeln('Введите кол-во спортсменов:');
  readln(N);
  maxFml := '';
  maxRsltt := 1000000;
  for i:=1 to N do
  begin
   writeln('Введите результат и фамилию спортсмена:');
    readln(lchshrsltt);
    readln(fmlpbdtl);
  if lchshrsltt < maxRsltt then
  begin
    maxRsltt := lchshrsltt;
    maxFml := fmlpbdtl;
  end;
  end;
  writeln('Победил', ' ', maxFml, ' ', 'с результатом:', ' ', maxRsltt);
end.